script.js

```diff
@@ -1,12 +1,210 @@
+  });
+  rootElem.appendChild(selectMenu);
+}
+function createSearchBar() {
+  const rootElem = document.getElementById("root");
+
+  //search bar
+  const searchBar = document.createElement("input");
+  searchBar.id = "txtSearchBy";
+  searchBar.setAttribute("type", "string");
+  searchBar.addEventListener("input", function () {
+    drawEpiCard();
+  });
+
+  //search result
+  const searchResult = document.createElement("label");
+  searchResult.id = "searchResult";
+
+  // const resultP = document.createElement("p");
+  // resultP.innerHTML = `Got ${_localEpisodes.length} episode(s). The data has (originally) come from <a href= "https://www.tvmaze.com/">TVMaze.com</a><br>`;
+  // //rootElem.innerHTML = `Got ${_localEpisodes.length} episode(s). The data has (originally) come from <a href= "https://www.tvmaze.com/">TVMaze.com</a><br>`;
+
+  // rootElem.insertBefore(resultP, rootElem.firstChild);
+  rootElem.appendChild(searchBar);
+  rootElem.appendChild(searchResult);
+  //drawEpiCard();
+}
+function clearCards() {
+  document
+    .querySelectorAll("#root div.grid-container")
+    .forEach((el) => el.remove());
+}
+function drawEpiCard() {
+  clearCards();
+  //get search words
+  const searchBar = document.getElementById("txtSearchBy");
+  let searchBy = searchBar.value.toLowerCase();
 
-function makePageForEpisodes(episodeList) {
+  //root
   const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+
+  //selected option
+  const selectMenu = document.getElementById("selName");
+  let selectedName = selectMenu.value;
+
+  //create container
+  const container = document.createElement("div");
+  container.className = "grid-container";
+
+  let found = 0;
+  _localEpisodes.forEach((episode) => {
+    if (
+      //ddl filter
+      (selectedName === "--Please Select--" ||
+        selectedName.includes(episode.name)) &&
+      //search bar filter
+      (!searchBy ||
+        episode.summary.toLowerCase().includes(searchBy) ||
+        episode.name.toLowerCase().includes(searchBy))
+    ) {
+      found++;
+      container.appendChild(getEpiCard(episode));
+    }
+  });
+  rootElem.appendChild(container);
+  const searchResult = document.getElementById("searchResult");
+  if (found > 0 && searchBy) {
+    searchResult.innerText = `Displaying ${found}/${_localEpisodes.length} episodes`;
+  } else {
+    searchResult.innerText = "";
+  }
+}
+function formatVal(val) {
+  return val.toString().padStart(2, "0");
+}
+function getEpiCard(episode) {
+  const divEl = document.createElement("div");
+  const titleEl = document.createElement("h1");
+  const imgEl = document.createElement("img");
+  divEl.appendChild(titleEl);
+  divEl.appendChild(imgEl);
+  divEl.className = "card ";
+  imgEl.className = "image";
+  titleEl.className = "title";
+  titleEl.innerText = `${episode.name} - S${formatVal(
+    episode.season
+  )}E${formatVal(episode.number)}`;
+  imgEl.src = episode.image.medium;
```

Comment:
there is a bug here for some of the shows (for example Alaska: The Last Frontier) - there is a console error of `Cannot read property 'medium' of null` - this is a good example of where it's always useful to think about what could go wrong when using someone else's data, for example what happens if there is no summary? Or image? It's always good to use `if` statements to handle cases like this :)